source/adapters/level_zero/adapter.cpp

```diff
@@ -385,8 +385,10 @@ ur_adapter_handle_t_::ur_adapter_handle_t_()
         }
       }
 
-      if (*GlobalAdapter->ZeInitResult == ZE_RESULT_SUCCESS ||
-          *GlobalAdapter->ZeInitDriversResult == ZE_RESULT_SUCCESS) {
+      if ((GlobalAdapter->ZeInitResult != std::nullopt &&
```

Comment:
Originally we used `GlobalAdapter->ZeResult` being empty to check whether initialization already happened. These other variables don't seem to be serving the same function. My suggestion would to not use std::optional for them, and set them initially to `ZE_RESULT_ERROR_UNINITIALIZED` or `ZE_RESULT_ERROR_UNKNOWN` or something.

Reply:
or, now that I thought about for a second, we can just use bools for these. I don't think we care about specific values apart from "SUCCESS".